source/extensions/filters/http/cache/cache_headers_utils.cc

```diff
@@ -0,0 +1,170 @@
+#include "extensions/filters/http/cache/cache_headers_utils.h"
+
+#include <array>
+#include <string>
+
+#include "envoy/common/time.h"
+
+#include "absl/algorithm/container.h"
+#include "absl/strings/ascii.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_split.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+// Utility functions used in RequestCacheControl & ResponseCacheControl
+namespace {
+OptionalDuration parseDuration(absl::string_view s) {
+  OptionalDuration duration;
+  // Strip quotation marks if any
+  if (s.size() > 1 && s.front() == '"' && s.back() == '"') {
+    s = s.substr(1, s.size() - 2);
+  }
+  long num;
+  if (absl::SimpleAtoi(s, &num) && num >= 0) {
+    // s is a valid string of digits representing a positive number
+    duration = std::chrono::seconds(num);
```

Comment:
should we have some kind of error-state return so that we can drop the entire request if the duration is malformed?

Reply:
The current behavior is that directives with a malformed duration are ignored. The RFC [https://httpwg.org/specs/rfc7234.html#delta-seconds] does not specify what should be done if a duration is malformed.

Reply:
Comments referencing the RFC when making policy decisions would be helpful.